.github/workflows/build.yml

```diff
@@ -31,11 +31,13 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install ruff
+          python -m pip install uv
```

Comment:
Nit: I'd suggest https://github.com/astral-sh/setup-uv for getting `uv` onto the actions runners.